cmd/gitserver/server/vcs_syncer.go

```diff
@@ -302,3 +308,141 @@ func (s *PerforceDepotSyncer) RemoteShowCommand(ctx context.Context, remoteURL *
+	repository, groupID, artifactID := decomposeMavenURL(remoteURL)
+	coursierVersions, err := coursier.ListVersions(ctx, repository, groupID, artifactID)
+	if err != nil {
+		return err
+	}
+
+	cmd := exec.CommandContext(ctx, "git", "show-branch")
+	dir.Set(cmd)
+	gitVersionsRaw, err := runWith(ctx, cmd, false, nil)
+	if err != nil {
+		return errors.Wrapf(err, "failed to git show-branch with output %q", string(gitVersionsRaw))
+	}
+	gitVersionsLines := strings.Split(string(gitVersionsRaw), "\n")
+
+	gitVersions := make(map[string]struct{})
+	for _, rawVersion := range gitVersionsLines {
+		gitVersions[versionPattern.FindStringSubmatch(rawVersion)[1]] = struct{}{}
+	}
+	var filteredCoursierVersions []string
+	for _, version := range coursierVersions {
+		if _, exists := gitVersions[version]; !exists {
+			filteredCoursierVersions = append(filteredCoursierVersions, version)
+		}
+	}
+
+	paths, err := coursier.FetchVersions(ctx, repository, groupID, artifactID, filteredCoursierVersions)
+
+	return s.commitJars(ctx, dir, paths, filteredCoursierVersions)
+}
+
+func (s MavenArtifactSyncer) commitJars(ctx context.Context, dir GitDir, paths, versions []string) error {
+	for idx, path := range paths {
+		cmd := exec.CommandContext(ctx, "git", "checkout", "--orphan", versions[idx])
+		dir.Set(cmd)
+		if output, err := runWith(ctx, cmd, false, nil); err != nil {
+			return errors.Wrapf(err, "failed to git checkout with output %q", string(output))
+		}
+
+		cmd = exec.CommandContext(ctx, "git", "reset")
+		dir.Set(cmd)
+		if output, err := runWith(ctx, cmd, false, nil); err != nil {
+			return errors.Wrapf(err, "failed to git reset with output %q", string(output))
+		}
+
+		cmd = exec.CommandContext(ctx, "git", "clean", "-d", "-f")
+		dir.Set(cmd)
+		if output, err := runWith(ctx, cmd, false, nil); err != nil {
+			return errors.Wrapf(err, "failed to git clean with output %q", string(output))
+		}
+
+		cmd = exec.CommandContext(ctx, "unzip", path, "-d", "./")
+		dir.Set(cmd)
+		if output, err := runWith(ctx, cmd, false, nil); err != nil {
+			return errors.Wrapf(err, "failed to unzip with output %q", string(output))
+		}
+
+		// TODO: codeintel, need some extra logic here to set up maven repo properly for indexing
+
+		cmd = exec.CommandContext(ctx, "git", "add", "*")
+		dir.Set(cmd)
+		if output, err := runWith(ctx, cmd, false, nil); err != nil {
+			return errors.Wrapf(err, "failed to git add with output %q", string(output))
+		}
+
+		cmd = exec.CommandContext(ctx, "git", "commit", "-m", versions[idx])
+		dir.Set(cmd)
+		if output, err := runWith(ctx, cmd, false, nil); err != nil {
+			return errors.Wrapf(err, "failed to git commit with output %q", string(output))
+		}
+	}
+
+	return nil
+}
+
+// RemoteShowCommand returns the command to be executed for showing remote.
+func (s MavenArtifactSyncer) RemoteShowCommand(ctx context.Context, remoteURL *vcs.URL) (cmd *exec.Cmd, err error) {
+	// TODO: we return nil
+	return nil, nil
```

Comment:
I'm pretty sure just do what `PerforceDepotSyncer` does is always safe as long as it is a Git repository:

```suggestion
	return exec.CommandContext(ctx, "git", "remote", "show", "./"), nil
```